apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -1442,13 +1531,43 @@ func (p *ProvisionerAPI) CharmProfileChangeInfo(machines params.Entities) (param
 	return params.ProfileChangeResults{Results: results}, nil
 }
 
-func machineChangeProfileChangeInfo(machine ProfileMachine, st ProfileBackend) (params.ProfileChangeResult, error) {
-	nothing := params.ProfileChangeResult{}
```

Comment:
calling this "ChangeProfileChangeInfo" makes me a bit concerned. "changed relative to what"?

Most operations we do, we want to get an idea of "this is the current state that we want" and then we compare it to the state we are currently at

the state we seem to want to compare to is lxc profile show (the api equivalent). Otherwise we risk half applying a profile and then we end up re applying it or never applying the second part.

I think that could help with issues about subordinates showing up unexpected, etc.

Reply:
Talked about this offline and it's on the list of things to address in the future.